packages/dds/tree/src/simple-tree/api/schemaFactoryRecursive.ts

```diff
@@ -159,37 +153,101 @@ export function createFieldSchemaUnsafe<
-				: unknown;
-			[NodeKind.Array]: T["info"] extends ImplicitAllowedTypes
-				? Iterable<InsertableTreeNodeFromImplicitAllowedTypes<T["info"]>>
-				: unknown;
-			[NodeKind.Map]: T["info"] extends ImplicitAllowedTypes
-				? Iterable<[string, InsertableTreeNodeFromImplicitAllowedTypes<T["info"]>]>
-				: unknown;
-		}[T["kind"]],
-		// ImplicitlyConstructable: recursive types are currently not implicitly constructable.
-		false,
-		// Info: What's passed to the method to create the schema. Constraining these here should be about as effective as if the actual constraints existed on the actual method itself.
-		{
-			[NodeKind.Object]: RestrictiveStringRecord<ImplicitFieldSchema>;
-			[NodeKind.Array]: ImplicitAllowedTypes;
-			[NodeKind.Map]: ImplicitAllowedTypes;
-		}[T["kind"]]
-	>,
+	T extends ValidateRecursiveSchemaTemplate<T>,
 > = true;
 
+/**
+ * @system @public
+ */
+export type ValidateRecursiveSchemaTemplate<T extends TreeNodeSchema> = TreeNodeSchema<
+	// Name: This validator places no restrictions on the name other than that it's a string (as required by TreeNodeSchemaClass).
+	string,
+	// NodeKind: These are the NodeKinds which currently can be used recursively.
+	NodeKind.Array | NodeKind.Map | NodeKind.Object,
+	// TNode: The produced node API. This is pretty minimal validation: more could be added if similar to how TInsertable works below if needed.
+	TreeNode & WithType<T["identifier"], T["kind"]>,
+	// TInsertable: What can be passed to the constructor. This should be enough to catch most issues with incorrect schema.
+	// These match whats defined in the recursive methods on `SchemaFactory` except they do not use `Unenforced`.
+	{
+		[NodeKind.Object]: T["info"] extends RestrictiveStringRecord<ImplicitFieldSchema>
+			? InsertableObjectFromSchemaRecord<T["info"]>
+			: unknown;
+		[NodeKind.Array]: T["info"] extends ImplicitAllowedTypes
+			? Iterable<InsertableTreeNodeFromImplicitAllowedTypes<T["info"]>>
+			: unknown;
+		[NodeKind.Map]: T["info"] extends ImplicitAllowedTypes
+			? Iterable<[string, InsertableTreeNodeFromImplicitAllowedTypes<T["info"]>]>
+			: unknown;
+		[NodeKind.Leaf]: unknown;
+	}[T["kind"]],
+	// ImplicitlyConstructable: recursive types are currently not implicitly constructable.
+	false,
+	// Info: What's passed to the method to create the schema. Constraining these here should be about as effective as if the actual constraints existed on the actual method itself.
+	{
+		[NodeKind.Object]: RestrictiveStringRecord<ImplicitFieldSchema>;
+		[NodeKind.Array]: ImplicitAllowedTypes;
+		[NodeKind.Map]: ImplicitAllowedTypes;
+		[NodeKind.Leaf]: unknown;
+	}[T["kind"]]
+>;
+
+/**
+ * Workaround for "Type instantiation is excessively deep and possibly infinite.ts" errors.
+ * @remarks
+ *
+ * Generally this workaround should be avoided if possible,
+ * especially for exported types as it is likely to result in issues when exporting or importing schema forcing the user to use the workaround as well.
```

Comment:
Nit (for readability)
```suggestion
 * especially for exported types, as it is likely to result in issues when exporting or importing schema where the user will be forced to use the workaround as well.
```